src/core/Makefile.am

```diff
@@ -368,6 +368,7 @@ HEADERS_COMMON                      = \
     thread/child_table.hpp            \
     thread/energy_scan_server.hpp     \
     thread/indirect_sender.hpp        \
+    thread/indirect_sender_frame_context.hpp \
```

Comment:
nit: align `\`

Alternatively, is this separate file necessary? Or can we merge with `indirect_sender.hpp`?

Reply:
We seem to need it to be a separate header file (to avoid include loops). I would also like it  to be in `indirect_sender.hpp` itself, but could not find a good to make it work.

- We want `IndirectSender` to be the one to define the context content, but `DataPollHandelr` be the class to provide the object/buffer for it.
- `indirect_sender.hpp` needs to include `data_poll_handler.hpp` (since it defines a member variable of type `DataPollHander` and it needs to know the full type definition).
- On other hand, the `data_poll_hanlder.hpp` needs to know the full `FrameContext` definition to be able to define member variable of this type within `DataPollHandler` class itself. 
- We need full types in both cases, so forward declerations won't work.
- So having the context be defined in a separate header file enable us to include it in `data_poll_hanlder.hpp` and avoid include header file loop.

Thoughts? Suggestions?